File: Dmhca1-main/extract_admin_courses.mjs
import fs from 'fs';
import readline from 'readline';

const transcriptFile = 'c:\\Users\\john\\AppData\\Roaming\\Code\\User\\workspaceStorage\\63e64f3523d8f17852f08381b21a7b74\\GitHub.copilot-chat\\transcripts\\643c312a-55ff-4dc5-bcdc-fb2413c12590.jsonl';

let latestCode = null;
let lineCount = 0;

const readStream = fs.createReadStream(transcriptFile, { encoding: 'utf8' });
const rl = readline.createInterface({
  input: readStream,
  crlfDelay: Infinity
});

rl.on('line', (line) => {
  lineCount++;
  try {
    const data = JSON.parse(line);
    if (data.data && data.data.content) {
      const content = data.data.content;
      if (content.includes('export const Route = createFileRoute') && content.includes('admin.courses')) {
        if (content.length > 5000) {
          console.log(`Found at line ${lineCount} with ${content.length} chars`);
          latestCode = content;
        }
      }
    }
  } catch (e) {
    // Ignore JSON parse errors
  }
});

rl.on('close', () => {
  console.log(`Read ${lineCount} lines`);
  if (latestCode) {
    console.log(`\nExtracting code...`);
    const codeBlockRegex = /```[a-z]*\n([\s\S]*?)\n```/g;
    let match;
    let largestBlock = '';
    let blockCount = 0;
    
    while ((match = codeBlockRegex.exec(latestCode)) !== null) {
      blockCount++;
      if (match[1].length > largestBlock.length) {
        largestBlock = match[1];
      }
    }
    
    if (largestBlock) {
      console.log(`Found ${blockCount} code blocks, largest: ${largestBlock.length} chars`);
      fs.writeFileSync('extracted_admin_courses.tsx', largestBlock, 'utf8');
      console.log('Saved to extracted_admin_courses.tsx');
    } else {
      console.log('No code blocks found with expected format');
    }
  } else {
    console.log('No matching content found');
  }
});

File: Dmhca1-main/extract_admin_courses.py
#!/usr/bin/env python3
import json
import re

transcript_file = r'c:\Users\john\AppData\Roaming\Code\User\workspaceStorage\63e64f3523d8f17852f08381b21a7b74\GitHub.copilot-chat\transcripts\643c312a-55ff-4dc5-bcdc-fb2413c12590.jsonl'

latest_code = None
latest_line = 0

with open(transcript_file, 'r', encoding='utf-8', errors='replace') as f:
    for line_num, line in enumerate(f, 1):
        try:
            data = json.loads(line)
            if 'data' in data and 'content' in data['data']:
                content = data['data']['content']
                # Look for code blocks with admin.courses or export const Route
                if 'export const Route = createFileRoute' in content and 'admin.courses' in content:
                    if len(content) > 5000:  # Large code block
                        print(f'Found at line {line_num}: {len(content)} chars')
                        latest_code = content
                        latest_line = line_num
        except:
            pass

if latest_code:
    print(f'\n\nExtracting code from line {latest_line}...')
    # Find all code blocks
    code_blocks = re.findall(r'```[a-z]*\n(.*?)```', latest_code, re.DOTALL)
    if code_blocks:
        print(f'Found {len(code_blocks)} code blocks')
        # Get the largest code block (should be the full file)
        largest_block = max(code_blocks, key=len)
        print(f'\nLargest block: {len(largest_block)} chars')
        with open('extracted_admin_courses.tsx', 'w', encoding='utf-8') as out:
            out.write(largest_block)
        print('Saved to extracted_admin_courses.tsx')
